lib/proxy/router.go

```diff
@@ -282,16 +285,31 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 				}
 			}
 		}
-
 	} else {
+		// DELETE IN 15.0.0: necessary for smoothing over v13 to v14 transition only.
+		// In v15 the connection problem error should always be returned here.
 		if !r.permitUnlistedDialing {
 			return nil, trace.ConnectionProblem(errors.New("connection problem"), "direct dialing to nodes not found in inventory is not supported")
```

Comment:
Not a part of this PR per se, but I think that we should also mention that the nodes can be registered with `teleport join openssh` to make it easier for people to fix that error when it happens.

Reply:
That's a good idea, but I have a docs PR open that will change the URL of the guide when merged so I'll change this error message in that PR.

Reply:
I didn't realize docs can do redirects (obvious in hindsight) so I improved the error message as you suggested.